lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,6 +162,24 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if (not isinstance(encoded_text, list)
+            or not isinstance(vocabulary, dict)
+            or not isinstance(end_of_word_token, str | None)):
+        return None
+
+    decoded_list = []
+    for number in encoded_text:
+        token_list = [token for token, value in vocabulary.items() if value == number]
+        for token in token_list:
+            if end_of_word_token:
+                if end_of_word_token in token:
+                    deleted_token = len(end_of_word_token)
+                    decoded_list += end_of_word_token[deleted_token:] + ' '
```

Comment:
strange code. Do you replace `end_of_word_token` with ' '?

Reply:
fixed